From ed1643951edf1f50b6f0a58e0e72c2a5fe69da31 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:03:18 -0400
Subject: [fs] btrfs: do not increment on bio_index one by one

Message-id: <1407179109-13714-312-git-send-email-sandeen@redhat.com>
Patchwork-id: 87270
O-Subject: [RHEL7.1 PATCH 311/422] Btrfs: do not increment on bio_index one by one
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Liu Bo <bo.li.liu@oracle.com>

commit d2cbf2a260ab18c833f07fda66e30c4d4344162e
Author: Liu Bo <bo.li.liu@oracle.com>
Date:   Tue Apr 29 13:07:58 2014 +0800

    Btrfs: do not increment on bio_index one by one

    'bio_index' is just a index, it's really not necessary to do increment
    one by one.

    Signed-off-by: Liu Bo <bo.li.liu@oracle.com>
    Reviewed-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/file-item.c b/fs/btrfs/file-item.c
index 9d84658..344b89c 100644
--- a/fs/btrfs/file-item.c
+++ b/fs/btrfs/file-item.c
@@ -281,10 +281,10 @@ static int __btrfs_lookup_bio_sums(struct btrfs_root *root,
 found:
   csum += count * csum_size;
   nblocks -= count;
+  bio_index += count;
   while (count--) {
    disk_bytenr += bvec->bv_len;
    offset += bvec->bv_len;
-   bio_index++;
    bvec++;
   }
  }
-- 
1.7.1